### ISSUE TITLE
mega linter does not show the same errors as the extension vscode

### ISSUE THREAD

**Issue by jsolas** [megalinter/megalinter#1296]
Hi, when parsing this file with npm-groovy-linter locally it shows me a number of errors and when I upload the code to the remote repository the mega linter tells me that this file has no errors

I have my rules defined in the .groovylintrc.json.

This is my github workflow setup

<img width="849" alt="image" src="https://user-images.githubusercontent.com/8138602/155402873-ddba9d2f-69a4-4d46-af40-3a4f881223ea.png">

This shows me on github:

<img width="832" alt="image" src="https://user-images.githubusercontent.com/8138602/155403426-2fe137f8-fff3-48bc-92bf-1ef42588342e.png">

Parsing this same file locally:
<img width="555" alt="image" src="https://user-images.githubusercontent.com/8138602/155403699-55781692-e0ef-4732-92bd-dbd02b681003.png">

please I appreciate your help, thank you

**Comment by nvuillam** [comment]
Where is defined your .groovylintrc.json ? At the root of the repository ?

What happens if you remove LINTER_RULES_PATH from your .mega-linter.yml file ? 

npm-groovy-lint / VsCode Groovy Lint are supposed to find themselves the config file at the root of the repo

**Comment by jsolas** [comment]
> .mega-linter.yml

Hello, thanks for your answer, if I have the groovylintrc.json defined in the root directory, now what I have different is that I don't have the mega linter configurations in the .mega-linter.yml file, I have them in my own yml file inside from the github workflow directory

<img width="1181" alt="image" src="https://user-images.githubusercontent.com/8138602/155749510-bbf27686-1e39-4991-b5c9-68ea29d7ac51.png">

**Comment by jsolas** [comment]
Why does the mega-linter takes this route?
![image](https://user-images.githubusercontent.com/8138602/155757513-69ffcab0-d45f-40b6-94a2-557a0f88425b.png)

**Comment by echoix** [comment]
The `/github/workspace/` seems like the absolute path to where your repo is cloned, either from the Github actions point of view or MegaLinter's container point of view. Note that it is different from the `.github` folder in your repo. What I see is that the configuration file used is in the root of your repo.

**Comment by jsolas** [comment]
> The `/github/workspace/` seems like the absolute path to where your repo is cloned, either from the Github actions point of view or MegaLinter's container point of view. Note that it is different from the `.github` folder in your repo. What I see is that the configuration file used is in the root of your repo.

Why does not show same errors as the extensions?

**Comment by echoix** [comment]
I'm curious to know about the versions of the linters used. When you are running it locally, what node.js and groovy version is used? Does it match the ones used in the MegaLinter GitHub action?

And I never used groovy personally, but is there a way to make sure the good configuration file is used when running locally? By that I want to rule out the possibility that a groovy configuration file in your current directory (somewhere outside the repo directory) overrides the one you specify. 

Otherwise, is there a chance that the files in GitHub aren't up to date (committed) between your local repo and what is on GitHub?

**Comment by jsolas** [comment]
>

**Comment by jsolas** [comment]
> I'm curious to know about the versions of the linters used. When you are running it locally, what node.js and groovy version is used? Does it match the ones used in the MegaLinter GitHub action?
> 
> And I never used groovy personally, but is there a way to make sure the good configuration file is used when running locally? By that I want to rule out the possibility that a groovy configuration file in your current directory (somewhere outside the repo directory) overrides the one you specify.
> 
> Otherwise, is there a chance that the files in GitHub aren't up to date (committed) between your local repo and what is on GitHub?

Local versions:
<img width="363" alt="image" src="https://user-images.githubusercontent.com/8138602/155786193-093cdd1b-e0d1-4885-bc4e-34f08f641890.png">

Remote repository:
![image](https://user-images.githubusercontent.com/8138602/155786493-a3df4510-9778-4f64-a556-d3e4ef10b4fa.png)
I don't know how to see the versions of node or groovy in the remote repository

**Comment by jsolas** [comment]
@echoix @nvuillam thanks for your answers, something curious that maybe can give you a clue is that this route is delivered by the mega linter log to a file if I analyze it locally with `npm-groovy-lint -f "relative_path"`, it does not find errors in local, example:
![image](https://user-images.githubusercontent.com/8138602/155796248-de7ecd48-4917-4e02-84c1-2dd83ddf7566.png)

npm-groovy-lint -f "grails-app/services/ctrl/UtilService.groovy" in this way it does not find errors, I have to do npm-groovy-lint -f "**/UtilService.groovy"

**Comment by echoix** [comment]
Could you try to also specify, in two tests, the `--path` argument, and the `--config` argument, like in the example command : 

```

npm-groovy-lint --path "./path/to/my/groovy/files" --files "**/*.groovy" --config "./config/codenarc/.groovylintrcCustom.js" --loglevel warning --output txt

```

https://nvuillam.github.io/npm-groovy-lint/#usage


This is to make sure that the full path of the config file is used, without ambiguity. 


Before running the command locally, are you `cd` ed in the repo's directory?


Does the messages in the cropped screenshots mentions errors in other files that "aren't accessible" when running either locally or in GH action?

**Comment by jsolas** [comment]
https://github.com/megalinter/megalinter/blob/bcbd8e2373022ecd1fb7324e352dc1f5ba03ca00/megalinter/linters/GroovyNpmGroovyLintLinter.py#L28 @nvuillam @echoix  my theory is that here the correct relative or absolute paths are not being received and for this reason it does not find errors

**Comment by cronlabspl** [comment]
Well, sorry for hijacvking issue, but am experiencing almost same behavior.

Am developing labspl/forem-ng, installe3d ML, while retaining other CQ tools active.

Observed inconsistency: ML fired warning in places where other CQ tool kept silent.

**Comment by nvuillam** [comment]
So, to summarize:
- Aggregating linters is a bad idea
- You use SuperLinter , which aggregates linters :)

Anyone is free to use any open source app they like, it's your full right to dislike MegaLinter, but in that case just ignore it instead of acting like an instagram hater in the repo issue ,thanks ^^

**Comment by cronlabspl** [comment]
> So, to summarize:
> 
> * Aggregating linters is a bad idea
> * You use SuperLinter , which aggregates linters :)
> 
> Anyone is free to use any open source app they like, it's your full right to dislike MegaLinter, but in that case just ignore it instead of acting like an instagram hater in the repo issue ,thanks ^^

As of first one: you misunderstood me; I said that **its of no sense to create yet another linters aggregating software**, not that aggregating itself is bad idea. Creating yet another just adds unnecesary mess I talked about in earlier post.

Not to mention that creating yet another aggregated linter adds SOPS everywhere.

**Comment by jsolas** [comment]
Hello @nvuillam  , I think the discussion deviated from the main topic, any suggestions, with my problem?

**Comment by nvuillam** [comment]
you're right, the troll changed the initial subject ^^
The solving would probably be in npm-groovy-lint (that I'm almost solo-maintainer), so I need to find some time to investigate
